academy/exchange/cloud/authenticate.py

```diff
@@ -69,30 +73,51 @@ class GlobusAuthenticator:
         audience: Intended audience of the token. This should typically be
             the resource server of the the token was issued for. E.g.,
             the UUID of the ProxyStore Relay Server application.
-        auth_client: Optional confidential application authentication client
-            which is used for introspecting client tokens.
     """
 
+    TOKEN_CACHE_LIMIT: int = 1024
+    TOKEN_TTL: int = 60
+
     def __init__(
         self,
         client_id: str | None = None,
         client_secret: str | None = None,
         *,
         audience: str = AcademyExchangeScopes.resource_server,
-        auth_client: globus_sdk.ConfidentialAppAuthClient | None = None,
     ) -> None:
-        self.auth_client = (
-            globus_sdk.ConfidentialAppAuthClient(
-                client_id=str(client_id),
-                client_secret=str(client_secret),
-            )
-            if auth_client is None
-            else auth_client
-        )
-        self.auth_client_lock = Lock()
+        self._local_data = threading.local()
+        self.executor = ThreadPoolExecutor()
```

Comment:
Can you set the `thread_name_prefix` on this? It helps when debugging to have the threads named.